runtime/libpgmath/lib/common/arm64intrin.h

```diff
@@ -0,0 +1,2761 @@
+    for (unsigned j = 0; j < 32; ++j) {
+      if (c.xi[i] & (1 << j)) {
+        if (b.xi[i] & (1 << j))
+          r.xi[i] |= (b.xi[i] & (1 << j));
+        else
+          r.xi[i] &= ~(b.xi[i] & (1 << j));
+      } else {
+        if (a.xi[i] & (1 << j))
+          r.xi[i] |= (a.xi[i] & (1 << j));
+        else
+          r.xi[i] &= ~(a.xi[i] & (1 << j));
+      }
+    }
+  }
+
+  return r;
+}
+
+static inline __m128d
+__attribute__((always_inline))
+vec_sel(const __m128d& a, const __m128d& b, const __m128d& c)
+{
+  __m128d r;
+
+  for (unsigned i = 0; i < 4; ++i) {
+    for (unsigned j = 0; j < 32; ++j) {
+      if (c.xi[i] & (1 << j)) {
+        if (b.xi[i] & (1 << j))
+          r.xi[i] |= (b.xi[i] & (1 << j));
+        else
+          r.xi[i] &= ~(b.xi[i] & (1 << j));
+      } else {
+        if (a.xi[i] & (1 << j))
+          r.xi[i] |= (a.xi[i] & (1 << j));
+        else
+          r.xi[i] &= ~(a.xi[i] & (1 << j));
+      }
+    }
+  }
+
+  return r;
+}
+
+static inline __m128i
+__attribute__((always_inline))
+vec_sel(const __m128i& a, const __m128i& b, const __m128i& c)
+{
+  __m128i r;
+
+  for (unsigned i = 0; i < 4; ++i) {
+    for (unsigned j = 0; j < 32; ++j) {
+      if (c.xi[i] & (1 << j)) {
+        if (b.xi[i] & (1 << j))
+          r.xi[i] |= (b.xi[i] & (1 << j));
+        else
+          r.xi[i] &= ~(b.xi[i] & (1 << j));
+      } else {
+        if (a.xi[i] & (1 << j))
+          r.xi[i] |= (a.xi[i] & (1 << j));
+        else
+          r.xi[i] &= ~(a.xi[i] & (1 << j));
+      }
+    }
+  }
+
+  return r;
+}
+
+static inline void
+__attribute__((always_inline))
+vec_sti(const __m128i& vi, int v, unsigned char* t)
+{
+  __m128i vix = v ? vi + v : vi;
+  unsigned char* x = vix.c;
+
+  t[0] = x[0];
+  t[1] = x[1];
+  t[2] = x[2];
+  t[3] = x[3];
+  t[4] = x[4];
+  t[5] = x[5];
+  t[6] = x[6];
+  t[7] = x[7];
+  t[8] = x[8];
+  t[9] = x[9];
+  t[10] = x[10];
+  t[11] = x[11];
+  t[12] = x[12];
+  t[13] = x[13];
+  t[14] = x[14];
+  t[15] = x[15];
+}
+
+static inline void
+__attribute__((always_inline))
+vec_stf(const __m128& vf, int v, unsigned char* t)
+{
+  __m128 vfx = v ? vf + v : vf;
+  unsigned char* x = vfx.c;
+
+  t[0] = x[0];
+  t[1] = x[1];
+  t[2] = x[2];
+  t[3] = x[3];
+  t[4] = x[4];
+  t[5] = x[5];
+  t[6] = x[6];
+  t[7] = x[7];
+  t[8] = x[8];
+  t[9] = x[9];
+  t[10] = x[10];
+  t[11] = x[11];
+  t[12] = x[12];
+  t[13] = x[13];
+  t[14] = x[14];
+  t[15] = x[15];
+}
+
+static inline void
+__attribute__((always_inline))
+vec_std(const __m128d& vd, int v, unsigned char* t)
+{
+  __m128d vdx = v ? vd + v : vd;
+  unsigned char* x = vdx.c;
+
+  t[0] = x[0];
+  t[1] = x[1];
+  t[2] = x[2];
+  t[3] = x[3];
+  t[4] = x[4];
+  t[5] = x[5];
+  t[6] = x[6];
+  t[7] = x[7];
+  t[8] = x[8];
+  t[9] = x[9];
+  t[10] = x[10];
+  t[11] = x[11];
+  t[12] = x[12];
+  t[13] = x[13];
+  t[14] = x[14];
+  t[15] = x[15];
+}
+
+/*
+ * No corresponding Altivec intrinsic to generate a scalar mask
```

Comment:
Altivec references left throughout this header.

Reply:
Yes, because this is an implementation of the Altivec interfaces.